Firestore/Example/Tests/Local/FSTLevelDBMigrationsTests.mm

```diff
@@ -199,6 +203,111 @@ - (void)testDropsTheQueryCacheWithThousandsOfEntries {
   }
 }
 
+- (void)testRemovesMutationBatches {
+  std::string emptyBuffer;
+  DocumentKey testWriteFoo = DocumentKey::FromPathString("docs/foo");
+  DocumentKey testWriteBar = DocumentKey::FromPathString("docs/bar");
+  DocumentKey testWriteBaz = DocumentKey::FromPathString("docs/baz");
+  DocumentKey testWritePending = DocumentKey::FromPathString("docs/pending");
+  // Do everything up until the mutation batch migration.
+  LevelDbMigrations::RunMigrations(_db.get(), 3);
+  // Set up data
+  {
+    LevelDbTransaction transaction(_db.get(), "Setup Foo");
+    // User 'foo' has two acknowledged mutations and one that is pending.
+    FSTPBMutationQueue *fooQueue = [[FSTPBMutationQueue alloc] init];
```

Comment:
Optional:

With a minor helper in here I think you could make these use nanopb firestore_client_MutationQueue messages instead. Here would be:

```
firestore_client_MutationQueue foo_queue{};
foo_queue.last_acknowledged_batch_id = 2;
std::string foo_key = LevelDbMutationQueueKey::Key("foo");
transaction.put(foo_key, Write(firestore_client_MutationQueue_fields, &foo_queue));
```
The helper would look like:
```
std::string Write(const pb_field_t fields[], const void* src_struct) {
  std::string result;
  Writer writer = Writer::Wrap(&result);
  writer.WriteNanopbMessage(fields, src_struct);
  return result;
}

Reply:
I initially tried including the nanopb headers here, but I ran into a build issue of some headers not being found. So, I ended up going w/ the ObjC protos like the rest of the file. Sadly, it looks like we can't port this test yet, as it has a bunch of ObjC dependencies.